### PR TITLE
EMTF add DQM plots for unconstrained pT and dxy

### PR DESCRIPTION
**Pull request by elledesa**
#### PR description:

This PR adds DQM plots using unconstrained pT and dxy. These plots will be useful for monitoring now that the neural network is working at P5.

This PR will be back-ported to 13_0_X and 13_1_X to be used online.

#### PR validation:

Validated by `runTheMatrix.py -l limited -i all --ibeos`  and everything looks good.

### CODE REVIEW

**Comment by emanueleusai** [comment]
type l1t

**Comment by emanueleusai** [comment]
please test

**Comment by perrotta** [comment]
please test

**Comment by emanueleusai** [comment]
+1

File: DQM/L1TMonitor/src/L1TStage2EMTF.cc
Comment (perrotta) [inline DQM/L1TMonitor/src/L1TStage2EMTF.cc:1050]: Why integers the two pts?
In addition, please make them constexpr (including quality), and move them outside the loop

**Comment by rappoccio** [comment]
hold

Place on explicit hold until comments from Andrea are addressed.

**Comment by eyigitba** [comment]
I can have a look at this if @lutuck is unable to apply the changes. It's not urgent and can stay on hold in the meantime

**Comment by menglu21** [comment]
Hi, it seems Lucas is unable to work on this, could you please close this PR and the backports, I'll make a new PR and backports with the comments addressed.

**Comment by perrotta** [comment]
> Hi, it seems Lucas is unable to work on this, could you please close this PR and the backports, I'll make a new PR and backports with the comments addressed.

Thank you @menglu21 : please go ahead with the PR, we can centrally close this one once the replacement is there

**Comment by emanueleusai** [comment]
please test

**Comment by emanueleusai** [comment]
+1

File: DQM/L1TMonitor/interface/L1TStage2EMTF.h
Comment (emanueleusai) [inline DQM/L1TMonitor/interface/L1TStage2EMTF.h:55]: remove comment

File: DQM/L1TMonitor/interface/L1TStage2EMTF.h
Comment (emanueleusai) [inline DQM/L1TMonitor/interface/L1TStage2EMTF.h:56]: remove comment

File: DQM/L1TMonitor/interface/L1TStage2EMTF.h
Comment (emanueleusai) [inline DQM/L1TMonitor/interface/L1TStage2EMTF.h:57]: remove comment

File: DQM/L1TMonitor/interface/L1TStage2EMTF.h
Comment (emanueleusai) [inline DQM/L1TMonitor/interface/L1TStage2EMTF.h:58]: remove comment

File: DQM/L1TMonitor/interface/L1TStage2EMTF.h
Comment (emanueleusai) [inline DQM/L1TMonitor/interface/L1TStage2EMTF.h:60]: remove comment

File: DQM/L1TMonitor/interface/L1TStage2EMTF.h
Comment (emanueleusai) [inline DQM/L1TMonitor/interface/L1TStage2EMTF.h:62]: remove comment

File: DQM/L1TMonitor/interface/L1TStage2EMTF.h
Comment (emanueleusai) [inline DQM/L1TMonitor/interface/L1TStage2EMTF.h:64]: remove comment

File: DQM/L1TMonitor/interface/L1TStage2EMTF.h
Comment (emanueleusai) [inline DQM/L1TMonitor/interface/L1TStage2EMTF.h:80]: remove comment

File: DQM/L1TMonitor/interface/L1TStage2EMTF.h
Comment (emanueleusai) [inline DQM/L1TMonitor/interface/L1TStage2EMTF.h:81]: remove comment

File: DQM/L1TMonitor/src/L1TStage2EMTF.cc
Comment (emanueleusai) [inline DQM/L1TMonitor/src/L1TStage2EMTF.cc:179]: remove comment

File: DQM/L1TMonitor/src/L1TStage2EMTF.cc
Comment (emanueleusai) [inline DQM/L1TMonitor/src/L1TStage2EMTF.cc:185]: remove comment

File: DQM/L1TMonitor/src/L1TStage2EMTF.cc
Comment (emanueleusai) [inline DQM/L1TMonitor/src/L1TStage2EMTF.cc:227]: remove comment

File: DQM/L1TMonitor/src/L1TStage2EMTF.cc
Comment (emanueleusai) [inline DQM/L1TMonitor/src/L1TStage2EMTF.cc:266]: remove comment

File: DQM/L1TMonitor/src/L1TStage2EMTF.cc
Comment (emanueleusai) [inline DQM/L1TMonitor/src/L1TStage2EMTF.cc:268]: remove comment

File: DQM/L1TMonitor/src/L1TStage2EMTF.cc
Comment (emanueleusai) [inline DQM/L1TMonitor/src/L1TStage2EMTF.cc:292]: remove comment

File: DQM/L1TMonitor/src/L1TStage2EMTF.cc
Comment (emanueleusai) [inline DQM/L1TMonitor/src/L1TStage2EMTF.cc:774]: remove comment

File: DQM/L1TMonitor/src/L1TStage2EMTF.cc
Comment (emanueleusai) [inline DQM/L1TMonitor/src/L1TStage2EMTF.cc:781]: remove comment

File: DQM/L1TMonitor/src/L1TStage2EMTF.cc
Comment (emanueleusai) [inline DQM/L1TMonitor/src/L1TStage2EMTF.cc:1039]: change to constexpr float

Comment (emanueleusai) [reply DQM/L1TMonitor/src/L1TStage2EMTF.cc:1039]: and 22.0

File: DQM/L1TMonitor/src/L1TStage2EMTF.cc
Comment (emanueleusai) [inline DQM/L1TMonitor/src/L1TStage2EMTF.cc:1040]: change to constexpr float and 10.0

File: DQM/L1TMonitor/src/L1TStage2EMTF.cc
Comment (emanueleusai) [inline DQM/L1TMonitor/src/L1TStage2EMTF.cc:1060]: remove comment

File: DQM/L1TMonitor/src/L1TStage2EMTF.cc
Comment (emanueleusai) [inline DQM/L1TMonitor/src/L1TStage2EMTF.cc:1061]: remove comment

File: DQM/L1TMonitor/src/L1TStage2EMTF.cc
Comment (emanueleusai) [inline DQM/L1TMonitor/src/L1TStage2EMTF.cc:1073]: remove comment

File: DQM/L1TMonitor/src/L1TStage2EMTF.cc
Comment (emanueleusai) [inline DQM/L1TMonitor/src/L1TStage2EMTF.cc:1083]: remove comment

File: DQM/L1TMonitor/src/L1TStage2EMTF.cc
Comment (emanueleusai) [inline DQM/L1TMonitor/src/L1TStage2EMTF.cc:1271]: remove comment

File: DQM/L1TMonitor/src/L1TStage2EMTF.cc
Comment (emanueleusai) [inline DQM/L1TMonitor/src/L1TStage2EMTF.cc:1272]: remove comment

**Review by emanueleusai — CHANGES_REQUESTED** [review]
@lutuck, I believe @perrotta suggests to change singleMuPT and singleMuUPT from int to float because track pT is not an integer quantity in general.
Additionally could  you please remove the name, date tags? they are not needed and do not add any useful info to the code. Author and date of the changes are tracked by github already.

**Comment by lutuck** [comment]
@emanueleusai  Yes, I will make these changes now! Would you like singleMuPT and singleMuUPT to still be constexpr float, or just float?

**Comment by emanueleusai** [comment]
constexpr float is fine

**Comment by emanueleusai** [comment]
please test

**Comment by perrotta** [comment]
unhold
@lutuck there are still a few comments from @emanueleusai to be implemented, see https://github.com/cms-sw/cmssw/pull/42155#pullrequestreview-1578585475 (I'll let Emanuele decide whether they are relevant or not)

**Comment by emanueleusai** [comment]
Yes, @lutuck please remove the comments with your name and date from the code. Functionally speaking the PR is ok now, but it is a matter of code cleanliness and readability.
I do not think each line you change should be commented with the date and name of the author. This is tracked by github already.

**Comment by lutuck** [comment]
@emanueleusai I believe I updated the PR to have the comments removed for the reasons you stated. Please let me know if you're still seeing this issue and I can take another look, or if there are any other modifications you'd like me to make.

**Comment by perrotta** [comment]
> @emanueleusai I believe I updated the PR to have the comments removed for the reasons you stated. Please let me know if you're still seeing this issue and I can take another look, or if there are any other modifications you'd like me to make.

@lutuck please have a look at all comments listed in https://github.com/cms-sw/cmssw/pull/42155#pullrequestreview-1578585475 that you have not yet implemented

**Comment by lutuck** [comment]
@perrotta the most recent commit (2 days ago) removed the comments and changed ints to constexpr floats. Please let me know if you're not seeing that on your end, and I can try to update the PR again. Please let me know as well if there are any other changes you would like me to implement.

**Comment by perrotta** [comment]
@lutuck please open the "Files changed" tab in this PR in github and go through the several other comments of @emanueleusai that you did not apply yet

**Comment by lutuck** [comment]
@perrotta I apologize for the confusion, I have now resolved the other comments for the header file. Please let me know if there are any other changes you'd like to make for this PR.

**Comment by perrotta** [comment]
@lutuck I still see some unanswered comment in https://github.com/cms-sw/cmssw/pull/42155/files
Can't you also see them?

**Comment by syuvivida** [comment]
+1

**Comment by syuvivida** [comment]
@lutuck 
 We don't seem to find the 13_2_X version of this PR. Could you create a backport to 13_2_X? For the time being, we are adding this PR on top of 13_2_3 and 13_2_4 but it would be easier for DQM if this PR is merged in one of the 13_2_X releases in the coming future. Thanks!!

**Comment by eyigitba** [comment]
@syuvivida , I think @lutuck won't be able to help with this. I can handle the backport to 13_2_X tomorrow.

**Pull request merged**
